quic/s2n-quic-qns/src/xdp.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Result;
+use aya::{
+    maps::{HashMap, MapData, XskMap},
+    programs, Bpf,
+};
+use s2n_quic::provider::io::{
+    self,
+    xdp::{
+        bpf, encoder,
+        if_xdp::{self, XdpFlags},
+        io::{self as xdp_io},
+        ring, socket, syscall, umem, Provider,
+    },
+};
+use std::{ffi::CString, net::SocketAddr, os::unix::io::AsRawFd, sync::Arc};
+use structopt::StructOpt;
+use tokio::{io::unix::AsyncFd, net::UdpSocket};
+
+#[derive(Debug, StructOpt)]
+pub struct Xdp {
+    #[structopt(long, default_value = "lo")]
+    interface: String,
+
+    #[structopt(long, default_value = "2048")]
+    tx_queue_len: u32,
+
+    #[structopt(long, default_value = "2048")]
+    rx_queue_len: u32,
+
+    #[structopt(long, default_value = "4096")]
+    frame_size: u32,
```

Comment:
Is there any guidance you could add around these defaults and when it might make sense to override them?

Reply:
The default values came from the kernel tests: https://elixir.bootlin.com/linux/latest/source/tools/testing/selftests/bpf/xsk.h#L185. I'm still playing around with how they affect performance so I'm not sure I can offer a ton of guidance right now on how to change them. But the same general rules apply, deeper queues can increase latency but also increase throughput.